src/plugins/dist_matrixops/dist_identity.cpp

```diff
@@ -0,0 +1,310 @@
+                    Other options are `row` or `column` tiling.
+                dtype (string, optional): the data-type of the returned array,
+                    defaults to 'float'.
+
+            Returns:
+
+            A part of an array of size 'size', which has the tile index of
+             'tile_index'.)")};
+
+    ///////////////////////////////////////////////////////////////////////////
+    dist_identity::dist_identity(
+        execution_tree::primitive_arguments_type&& operands,
+        std::string const& name, std::string const& codename)
+      : primitive_component_base(std::move(operands), name, codename)
+    {
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    namespace detail {
+        static std::atomic<std::size_t> const_count(0);
+        std::string generate_identity_name(std::string&& given_name)
+        {
+            if (given_name.empty())
+            {
+                return "identity_array_" + std::to_string(++const_count);
+            }
+
+            return std::move(given_name);
+        }
+    }    // namespace detail
+
+    ///////////////////////////////////////////////////////////////////////////
+    template <typename T>
+    execution_tree::primitive_argument_type dist_identity::dist_identity_helper(
+        std::int64_t&& sz, std::uint32_t const& tile_idx,
+        std::uint32_t const& numtiles, std::string&& given_name,
+        std::string const& tiling_type) const
+    {
+        using namespace execution_tree;
+
+        std::size_t size = static_cast<std::size_t>(sz);
+
+        std::int64_t row_start, column_start;
+        std::size_t row_size, column_size;
+        std::uint32_t const row_dim = size;
+        std::uint32_t const column_dim = size;
+
+        std::tie(row_start, column_start, row_size, column_size) =
+            tile_calculation::tile_calculation_2d(
+                tile_idx, row_dim, column_dim, numtiles, tiling_type);
+
+        tiling_information_2d tile_info(
+            tiling_span(row_start, row_start + row_size),
+            tiling_span(column_start, column_start + column_size));
+
+        locality_information locality_info(tile_idx, numtiles);
+        annotation locality_ann = locality_info.as_annotation();
+
+        std::string base_name =
+            detail::generate_identity_name(std::move(given_name));
+
+        annotation_information ann_info(
+            std::move(base_name), 0);    //generation 0
+
+        auto attached_annotation =
+            std::make_shared<annotation>(localities_annotation(locality_ann,
+                tile_info.as_annotation(name_, codename_), ann_info, name_,
+                codename_));
+
+        blaze::DynamicMatrix<T> m(row_size, column_size, 0.0);
+        if (tiling_type == "row")
+        {
+            for (std::size_t i = 0; i != row_size; ++i)
+            {
+                std::size_t j = i + row_start;
+                m(i, j) = 1.0;
```

Comment:
I am not sure but I think we might be able to use a `blaze::band`. Please refer to `eye_operation`

Reply:
Changed to blaze::band.